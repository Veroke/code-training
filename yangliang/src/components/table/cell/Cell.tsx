import { defineComponent, ExtractPropTypes, onErrorCaptured, onMounted } from "vue";
import { addLog } from "../../../log/log";
import './cell.css';

export const cellProps = {
    component: {
      type: String,
      default: 'td'
    },
    column: {
      type: Object,
      default: {}
    },
    record: {
      type: Object
    },
    sorter: {
        type: Function,
        default: () => {}
    }
  }
  
export type CellProps = ExtractPropTypes<typeof cellProps>;

export default defineComponent({
    name: "Cell",
    props: cellProps,
    setup(props: CellProps, { attrs, emit, slots }) {
        onMounted(()=>{
            addLog({
              level: "info",
              message: {
                component: "Cell",
                props,
                msg: 'Some One Use Cell Component'
              }
            })
          })
        
          onErrorCaptured((err) => {
            addLog({
              level: "error",
              message: {
                component: "Cell",
                props,
                msg: err
              }
            })
          })

        const Component: any = props.component;
        const cellattrs = {
            class: `${Component}-cell cell`
        }
        return () => {
          return (
            <Component { ...cellattrs }>
              {props.record ? props.record[props.column.dataKey] : props.column.title }
              { 
                  (!props.record && props.column.sorter) && <span class="cell-order">
                      <svg onClick={()=>{
                          props.sorter(props.column.dataKey, true)
                      }} focusable="false" class="asc-order" data-icon="caret-up" width="1em" height="1em" fill="currentColor" aria-hidden="true" viewBox="0 0 1024 1024">
                          <path d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z">
                          </path>
                      </svg>
                      <svg onClick={()=>{
                          props.sorter(props.column.dataKey, false)
                      }} focusable="false" class="desc-order" data-icon="caret-down" width="1em" height="1em" fill="currentColor" aria-hidden="true" viewBox="0 0 1024 1024">
                          <path d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z">
                          </path>
                      </svg>
                  </span>
              }
            </Component>
          );
      };
    },
});