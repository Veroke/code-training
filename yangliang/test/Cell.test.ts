import { shallowMount } from "@vue/test-utils";
import { expect, test } from "vitest";
import Cell from "../src/components/table/cell/Cell";

test("mount component", () => {
    const wrapper = shallowMount(Cell, {
      props: {
        column:[
          {
            title: '姓名',
            dataKey: 'name',
            key: 'name',
          }
        ],
        record: {
            name: '张三',
            age: 18,
            address: '深圳市南山区',
        },
        component: 'td',
        sorter: () => {}
      },
    });
  
    expect(wrapper.html()).toMatchSnapshot();
});
  

