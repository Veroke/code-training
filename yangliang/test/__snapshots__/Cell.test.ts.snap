// Vitest Snapshot v1

exports[`mount component 1`] = `
""<td class=\\"td-cell cell\\">张三
     <!---->
   </td>""
`;
